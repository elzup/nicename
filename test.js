import m from './src'

function findById(res, id) {
  return res.find(jr => jr.info.id === id)
}

test('length_short', () => {
  const id = 'length_short'
  const s = findById(m('ssssss'), id)
  const a = findById(m('aaaa'), id)
  const c = findById(m('ccc'), id)
  expect([s, a, c]).toMatchSnapshot()
  expect([s, a, c].map(v => v.result.rank)).toEqual(['S', 'A', 'C'])
})

test('length_long', () => {
  const id = 'length_long'
  const s = findById(m('ssssss'), id)
  const a = findById(m('aaaaaaaaaa'), id)
  const b = findById(m('b'.repeat(20)), id)
  const c = findById(m('c'.repeat(21)), id)
  expect([s, a, b, c]).toMatchSnapshot()
  expect([s, a, b, c].map(v => v.result.rank)).toEqual(['S', 'A', 'B', 'C'])
})

test('string_order', () => {
  const id = 'string_order'
  const s = findById(m('_zzz'), id)
  const a = findById(m('elzup'), id)
  const b = findById(m('ooo'.repeat(20)), id)
  const c = findById(m('zombie'.repeat(21)), id)
  expect([s, a, b, c]).toMatchSnapshot()
  expect([s, a, b, c].map(v => v.result.rank)).toEqual(['S', 'A', 'B', 'C'])
})
