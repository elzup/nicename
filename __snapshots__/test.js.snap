// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ambiguity 1`] = `
Array [
  Object {
    "info": Object {
      "description": "あいまいなパターンが無いと良い",
      "id": "ambiguity",
      "notice": "「shi」や「si」は h を入れるかどうかを覚えにくい。",
    },
    "result": Object {
      "help": "入っていない",
      "rank": "S",
    },
  },
  Object {
    "info": Object {
      "description": "あいまいなパターンが無いと良い",
      "id": "ambiguity",
      "notice": "「shi」や「si」は h を入れるかどうかを覚えにくい。",
    },
    "result": Object {
      "help": "「si」と「shi」は間違えやすい",
      "rank": "A",
    },
  },
]
`;

exports[`length_long 1`] = `
Array [
  Object {
    "info": Object {
      "description": "長すぎないと良い",
      "id": "length_long",
      "notice": "表示しやすい。覚えやすい。",
    },
    "result": Object {
      "help": "6文字以下である",
      "rank": "S",
    },
  },
  Object {
    "info": Object {
      "description": "長すぎないと良い",
      "id": "length_long",
      "notice": "表示しやすい。覚えやすい。",
    },
    "result": Object {
      "help": "10文字以下である",
      "rank": "A",
    },
  },
  Object {
    "info": Object {
      "description": "長すぎないと良い",
      "id": "length_long",
      "notice": "表示しやすい。覚えやすい。",
    },
    "result": Object {
      "help": "20文字以下である",
      "rank": "B",
    },
  },
  Object {
    "info": Object {
      "description": "長すぎないと良い",
      "id": "length_long",
      "notice": "表示しやすい。覚えやすい。",
    },
    "result": Object {
      "help": "21文字以上である",
      "rank": "C",
    },
  },
]
`;

exports[`length_short 1`] = `
Array [
  Object {
    "info": Object {
      "description": "短すぎないと良い",
      "id": "length_short",
      "notice": "5文字以下だと登録できないサイトがある。",
    },
    "result": Object {
      "help": "6文字以上である",
      "rank": "S",
    },
  },
  Object {
    "info": Object {
      "description": "短すぎないと良い",
      "id": "length_short",
      "notice": "5文字以下だと登録できないサイトがある。",
    },
    "result": Object {
      "help": "4文字以上である",
      "rank": "A",
    },
  },
  Object {
    "info": Object {
      "description": "短すぎないと良い",
      "id": "length_short",
      "notice": "5文字以下だと登録できないサイトがある。",
    },
    "result": Object {
      "help": "3文字以下である",
      "rank": "C",
    },
  },
]
`;

exports[`sign_count 1`] = `
Array [
  Object {
    "info": Object {
      "description": "記号が少ないと良い",
      "id": "sign_count",
      "notice": "記号はユーザ名に使えない事が多い。ドメイン名やURLで使えないことが多い。",
    },
    "result": Object {
      "help": "入っていない",
      "rank": "S",
    },
  },
  Object {
    "info": Object {
      "description": "記号が少ないと良い",
      "id": "sign_count",
      "notice": "記号はユーザ名に使えない事が多い。ドメイン名やURLで使えないことが多い。",
    },
    "result": Object {
      "help": "1個ある",
      "rank": "A",
    },
  },
  Object {
    "info": Object {
      "description": "記号が少ないと良い",
      "id": "sign_count",
      "notice": "記号はユーザ名に使えない事が多い。ドメイン名やURLで使えないことが多い。",
    },
    "result": Object {
      "help": "2個ある",
      "rank": "B",
    },
  },
  Object {
    "info": Object {
      "description": "記号が少ないと良い",
      "id": "sign_count",
      "notice": "記号はユーザ名に使えない事が多い。ドメイン名やURLで使えないことが多い。",
    },
    "result": Object {
      "help": "3個以上ある",
      "rank": "C",
    },
  },
]
`;

exports[`string_order 1`] = `
Array [
  Object {
    "info": Object {
      "description": "名前順が早いと良い",
      "id": "string_order",
      "notice": "リストアップで上に出やすい。",
    },
    "result": Object {
      "help": "eより早い",
      "rank": "S",
    },
  },
  Object {
    "info": Object {
      "description": "名前順が早いと良い",
      "id": "string_order",
      "notice": "リストアップで上に出やすい。",
    },
    "result": Object {
      "help": "lより早い",
      "rank": "A",
    },
  },
  Object {
    "info": Object {
      "description": "名前順が早いと良い",
      "id": "string_order",
      "notice": "リストアップで上に出やすい。",
    },
    "result": Object {
      "help": "sより早い",
      "rank": "B",
    },
  },
  Object {
    "info": Object {
      "description": "名前順が早いと良い",
      "id": "string_order",
      "notice": "リストアップで上に出やすい。",
    },
    "result": Object {
      "help": "rより遅い",
      "rank": "C",
    },
  },
]
`;
